Websites/webkit.org/wp-content/themes/webkit/standards-positions.php

```diff
@@ -0,0 +1,266 @@
+<?php
+/**
+ * Template Name: Standards Positions
+ **/
+?>
+<?php get_header(); ?>
+<style>
+#content { overflow-x:auto }
+#content > h1 a { color:inherit }
+#content > p { width:80%; margin:0 auto }
+#content > p#filter-container { width:auto; font-size:2rem; margin-bottom:1em; padding:.2em; background:#fff; border: 1px solid hsl(0, 0%, 90.6%); }
+#content > p#filter-container > input { width:80%; margin:0 auto; border:0; display:block; box-sizing:border-box; font-size:2rem; background-position-x:0.5rem; background-position-y:1rem; color:#000 }
+#content > p#filter-container > span { width:80%; margin:0 auto; display:block; text-align:right; font-size:1.5rem; }
+#content > p#filter-container > span input { margin-left:.5em; margin-right:1em }
+#content > p#filter-container > span select { margin-left:.5em; width:8em }
+#content > table { table-layout:fixed; border-collapse:collapse; width:80%; margin:0 auto 50px }
+#content > table th { font-weight:bold; padding:1px; text-align:left }
+#content > table th:nth-child(1) { width:2.5em }
+#content > table th:nth-child(3) { width:9.5em }
+#content > table th:nth-child(4) { width:6.5em }
+#content > table th:nth-child(5) { width:8em }
+#content > table th:nth-child(6) { width:5em }
+#content > table th:nth-child(7) { width:1.5em }
+#content > table td { padding:1px }
+#content > table td:nth-child(4) { text-align:center }
+#content > table td:nth-child(4) > span { display:block; border-radius:15px/20px }
+#content > table td.oppose > span { background:#e40303; color:#fff }
+#content > table td.support > span { background:#008026; color:#fff }
+#content > table td.neutral >span { background:#ffed00; color:#000 }
+#content > table td.not-considering > span { background:#ff8c00; color:#000 }
+#content > table > tbody > tr { background:none }
+#content > table > tbody > tr:nth-child(even) { background: #fff }
+#content > table > tbody > tr:target { background:yellow }
+
+@media only screen and (max-width: 1200px) {
+  #content > table { width:90% }
+}
+
+@media only screen and (max-width: 1024px) {
+  #content > table th:nth-child(5),
+  #content > table td:nth-child(5) { display:none }
+  #content > table th:nth-child(6),
+  #content > table td:nth-child(6) { display:none }
+}
+
+@media only screen and (max-width: 800px) {
+  #content > table { width:98% }
+  #content > table th:nth-child(3),
+  #content > table td:nth-child(3) { display:none }
+}
+</style>
+
+<main id=content>
+  <h1><a href="/">Standards Positions</a></h1>
+
+  <p>Enable JavaScript for an interactive summary table of WebKit's standards positions. Failing that, browse the <a href="https://github.com/WebKit/standards-positions">standards-positions GitHub repository</a> directly.</p>
```

Comment:
Should this be wrapped in `<noscript>`? I would have expected something like:

```HTML
<table>
  ... all the usual table head stuff here disabled....
  <tbody>
    <noscript
      ><tr>
        <td colspan="x">
          <p>
            Enable JavaScript for an interactive summary table of WebKit's
            standards positions. Failing that, browse the
            <a href="https://github.com/WebKit/standards-positions"
              >standards-positions GitHub repository</a
            >
            directly.
          </p>
        </td>
      </tr>
    </noscript>
  </tbody>
</table>

```

Reply:
I'm not sure why this is better? It also doesn't parse the way you think it does unfortunately.

Reply:
> It also doesn't parse the way you think it does unfortunately.

yeah, I can see that in the DOM viewer, but it's presented correctly in every browser (i.e., the <noscript> being conditional on the script being disabled):
 
<img width="466" alt="Screenshot 2023-07-07 at 15 42 48" src="https://github.com/WebKit/WebKit/assets/870154/2ecdf056-c5f5-4f55-9acc-1a6a1103e597">

In this sense, every browser is doing exactly what it's supposed to do for this case. That's way better as you don't need to explain what the user is missing out on. They can literally see it...  

> I'm not sure why this is better? 

The reason it's better is that it puts the skeleton of the table back into the markup. Basically, all the JS from line 78-117 in  `createStandardsPositionsTable()` goes away (it's much easier to update the HTML than the JS). 

Consider, is it really necessary to do: 
```JS
 for (const heading of [
    "Specification",
    "Concerns",
    "Position",
    "Topics",
    "Venues",
  ]) {
    addElement(headerRow, "th", { scope: "col" }, heading);
  }
```
  
instead of just this, which is explicitly clear:

```HTML 
<tr>
  <th scope="col">Specification</th>
  <th scope="col">Concerns</th>
  <th scope="col">Position</th>
  <th scope="col">Topics</th>
  <th scope="col">Venues</th>
</tr>
```

Noting also that `addElement()` and `createElement()` adds indirection. 

It's also better because you can then just directly query, populate, and enable the bits of the table based on capability (in this case, JS). 

It's also better from an accessibility perspective because even with JS disabled, the users get to "see" what they are actually not getting (basically what is described in the `<p>`, including all the search input). 

Eventually, if we do populate this table with php, we will need the actual HTML structure there regardless.

Depending how fancy we want to get (and how many people really do hit this site with JS disabled), we could even do filtering serve-side for those folks if we wanted to, while also providing the client side code for local filtering. 

But even if we don't do that, having that can be represented statically/hard-coded not be JS will makes things easier for us to maintain.